scripts/pyqt5_to_pyqt6/pyqt5_to_pyqt6.py

```diff
@@ -775,20 +843,48 @@ def main(argv: Sequence[str] | None = None) -> int:
         action="store_true",
         help="Apply modifications that would break behavior on QGIS 3, hence code may not work on QGIS 3",
     )
+    parser.add_argument(
+        "--dry_run",
+        action="store_true",
+        help="Displays the changes that would be made, but does not modify any files.",
+    )
+    parser.add_argument(
+        "--logfile",
+        action="store",
+        help="Path to logging file",
+    )
+
     args = parser.parse_args(argv)
+    log_format = "%(message)s"
+
+    if args.logfile:
+        logging.basicConfig(
+            level=logging.DEBUG,
+            format=log_format,
+            filename=Path(args.logfile),
+            filemode="w",
+        )
+
+    else:
+        logging.basicConfig(level=logging.DEBUG, format=log_format)
 
     # get all scope for all qt enum
     for module in target_modules:
         for key, value in module.__dict__.items():
             get_class_enums(value)
 
     ret = 0
+
+    dry_run = args.dry_run if args.dry_run else False
+
+    if dry_run:
```

Comment:
```suggestion
    if args.dry_run:
```

`store_true` will default to False.

> 'store_true' and 'store_false' - These are special cases of 'store_const' used for storing the values True and False respectively. In addition, they create default values of False and True respectively.

Extracted from python doc